### PR TITLE
addSegmentationToView

### PR DESCRIPTION
**Pull request by peterBrxwn**

### CODE REVIEW

**Comment by peterBrxwn** [comment]
[[Flutter] AddSegmentationToView](https://trello.com/c/MYF6zTCj/1247-flutter-addsegmentationtoview)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,14 @@
+## xx.x.x
+* Added `addSegmentationToViewWithID` and `addSegmentationToViewWithName` methods to the countly views interface.
+
+* Updated underlying Android SDK version to 23.8.4
```
Comment (turtledreams) [inline CHANGELOG.md:4]: ... version to X.X.X

File: example/lib/page_views.dart [hunk 2]
```diff
@@ -51,6 +51,14 @@ class ViewsPage extends StatelessWidget {
     Countly.instance.views.updateGlobalViewSegmentation({'abcd': '123'});
   }
 
+  Future<void> addSegmentationToViewWithID() async {
+    await Countly.instance.views.addSegmentationToViewWithID(viewIDs[1], {'addSegmentationToViewWithID': 'page'});
+  }
+
+  Future<void> addSegmentationToViewWithName() async {
+    await Countly.instance.views.addSegmentationToViewWithName(viewIDs[1], {'addSegmentationToViewWithName': 'page'});
```
Comment (turtledreams) [inline example/lib/page_views.dart:59]: this expects name, not id

File: lib/views_internal.dart [hunk 3]
```diff
@@ -122,4 +122,30 @@ class ViewsInternal implements Views {
...
+    await _countlyState.channel.invokeMethod('addSegmentationToViewWithID', <String, dynamic>{'data': json.encode(args)});
+  }
+
+  @override
+  Future<void> addSegmentationToViewWithName(String viewName, Map<String, Object> segmentation) async {
+    if (!_countlyState.isInitialized) {
+      Countly.log('[Views] addSegmentationToViewWithName, "initWithConfig" must be called before "[Views] addSegmentationToViewWithName"', logLevel: LogLevel.ERROR);
+      return;
+    }
+    Countly.log('Calling "[Views] addSegmentationToViewWithName"');
```
Comment (turtledreams) [inline lib/views_internal.dart:145]: + with name:[$viewName]

File: lib/views_internal.dart [hunk 4]
```diff
@@ -122,4 +122,30 @@ class ViewsInternal implements Views {
...
     await _countlyState.channel.invokeMethod('stopAllViews', <String, dynamic>{'data': json.encode(args)});
   }
+
+  @override
+  Future<void> addSegmentationToViewWithID(String viewID, Map<String, Object> segmentation) async {
+    if (!_countlyState.isInitialized) {
+      Countly.log('[Views] addSegmentationToViewWithID, "initWithConfig" must be called before "[Views] addSegmentationToViewWithID"', logLevel: LogLevel.ERROR);
+      return;
+    }
+    Countly.log('Calling "[Views] addSegmentationToViewWithID"');
```
Comment (turtledreams) [inline lib/views_internal.dart:132]: + with view ID:[$viewID]

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (peterBrxwn) [reply CHANGELOG.md]: done

File: example/lib/page_views.dart (hunk 2 above)
Comment (peterBrxwn) [reply example/lib/page_views.dart]: done

File: lib/views_internal.dart (hunk 3 above)
Comment (peterBrxwn) [reply lib/views_internal.dart]: done

File: lib/views_internal.dart (hunk 4 above)
Comment (peterBrxwn) [reply lib/views_internal.dart]: done

File: lib/views_internal.dart [hunk 5]
```diff
@@ -122,4 +122,30 @@ class ViewsInternal implements Views {
...
+    await _countlyState.channel.invokeMethod('addSegmentationToViewWithID', <String, dynamic>{'data': json.encode(args)});
+  }
+
+  @override
+  Future<void> addSegmentationToViewWithName(String viewName, Map<String, Object> segmentation) async {
+    if (!_countlyState.isInitialized) {
+      Countly.log('[Views] addSegmentationToViewWithName, "initWithConfig" must be called before "[Views] addSegmentationToViewWithName"', logLevel: LogLevel.ERROR);
+      return;
+    }
+    Countly.log('Calling "[Views] addSegmentationToViewWithName" with view ID:[$viewName]');
```
Comment (turtledreams) [inline lib/views_internal.dart:145]: add a check here for viewName. if null or empty string return

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: lib/views_internal.dart (hunk 5 above)
Comment (peterBrxwn) [reply lib/views_internal.dart:153]: done

**Review by turtledreams — APPROVED** [review]

**Review by ijunaid — APPROVED** [review]
Please make sure to update the underlying SDK's and uncomment the relative code for it before merging that PR.

**Comment by turtledreams** [comment]
Yea now lets make it release ready

**Review by turtledreams — CHANGES_REQUESTED** [review]
Yea now lets make it release ready

**Pull request merged**